src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebHostEnvironmentTests.cs

```diff
@@ -58,16 +60,94 @@ public void ApplyEnvironmentSettingsUsesTheCorrectKeysAndProperties()
+        {
+            var environment = new WebHostEnvironment();
+            var tempPath = Path.GetTempPath();
+
+            environment.ContentRootPath = tempPath;
+            environment.WebRootPath = tempPath;
+
+            Assert.Equal(tempPath, environment.WebRootPath);
+            Assert.Equal(tempPath, environment.ContentRootPath);
+
+            Assert.IsType<PhysicalFileProvider>(environment.ContentRootFileProvider);
+            Assert.IsType<PhysicalFileProvider>(environment.WebRootFileProvider);
+
+            Assert.Equal(tempPath, ((PhysicalFileProvider)environment.ContentRootFileProvider).Root);
+            Assert.Equal(tempPath, ((PhysicalFileProvider)environment.WebRootFileProvider).Root);
+        }
+
+        [Fact]
+        public void RelativePathsAreMappedToFullPaths()
+        {
+            var environment = new WebHostEnvironment();
+            var relativeRootPath = "/some-relative-path";
+            var relativeSubPath = "/some-other-relative-path";
+            
+            // ContentRootPath is mapped relative to AppContext.BaseDirectory
+            environment.ContentRootPath = relativeRootPath;
+            Assert.Equal(Path.Combine(AppContext.BaseDirectory, relativeRootPath), environment.ContentRootPath);
+
+            // WebRootPath is mapped relative to ContentRootPath
+            environment.WebRootPath = relativeSubPath;
+            Assert.Equal(Path.Combine(AppContext.BaseDirectory, relativeRootPath, relativeSubPath), environment.WebRootPath);
+        }
+
+        [Fact]
+        public void CannotUnsetPaths()
+        {
+            var environment = new WebHostEnvironment();
+            var webRootPath = Path.GetTempPath();
+
+            environment.WebRootPath = webRootPath;
+
+            Assert.Equal(webRootPath, environment.WebRootPath);
+            Assert.Equal(webRootPath, ((PhysicalFileProvider)environment.WebRootFileProvider).Root);
+
+            // Setting WebRootPath to null no-ops
+            environment.WebRootPath = null;
+            Assert.Equal(webRootPath, environment.WebRootPath);
+            Assert.Equal(webRootPath, ((PhysicalFileProvider)environment.WebRootFileProvider).Root);
+
+            // Setting ContentRootPath to null falls back to current directory
+            environment.ContentRootPath = null;
+            Assert.Equal(AppContext.BaseDirectory, environment.ContentRootPath);
+            Assert.Equal(AppContext.BaseDirectory, ((PhysicalFileProvider)environment.ContentRootFileProvider).Root);
+        }
+
+        [Fact]
+        public void SetContentRootAfterRelativeWebRoot()
+        {
+            var environment = new WebHostEnvironment();
+            var webRootPath = "/some-relative-path";
+            var tempPath = Path.GetTempPath();
+
+            environment.WebRootPath = webRootPath;
+
+            Assert.Equal(Path.Combine(AppContext.BaseDirectory, webRootPath), environment.WebRootPath);
```

Comment:
This is only because it's using the test constructor, right? Since the ContentRoot hasn't been set to a relative path at this point, I think this should still be relative to `Directory.GetCurrentDirectory();`. I think we should change this test to just call the `WebHostEnvironment(Assembly? callingAssembly)` ctor to verify we don't regress that.